modules/iot-core/README.md

```diff
@@ -0,0 +1,112 @@
+
+Before executing, device certificates shall be created, for example using:
+
+```
+openssl req -x509 -newkey rsa:2048 -keyout rsa_private.pem -nodes -out rsa_cert.pem -subj "/CN=unused"
+```
+
+And then provision public certificate path in the devices yaml file following the convention device_id: device_cert
+
+
+```hcl
+module "iot-platform" {
+  source     = "./iot-core"
+  project_id = "my_project_id"
+  region = "europe-west1"
+  telemetry_pub_sub_topic_id = "telemetry_topic_id"
+  status_pub_sub_topic_id = "status_topic_id"
+  devices_yaml_file = "devices.yaml"
+}
+# tftest:modules=1:resources=2
+
+```
+
+Now, we can test sending telemetry messages from devices to our IoT Platform, for example using the MQTT demo client at https://github.com/googleapis/nodejs-iot/tree/main/samples/mqtt_example
+
+## Example with specific PubSub topics for custom MQTT topics
+
+If you need to match specific MQTT topics (eg, /temperature) into specific PubSub topics, you can use extra_telemetry_pub_sub_topic_ids for that, as in the following example:
+
+```hcl
+module "iot-platform" {
+  source     = "./iot-core"
+  project_id = "my_project_id"
+  region = "europe-west1"
+  telemetry_pub_sub_topic_id = "telemetry_topic_id"
+  status_pub_sub_topic_id = "statu_topic_id"
+  extra_telemetry_pub_sub_topic_ids = [{
+      "mqtt_topic" = "humidity"
+      "pub_sub_topic" =  "hum_topic_id"
+  },
+  {
+      "mqtt_topic" = "temperature"
+      "pub_sub_topic" =  "temp_topic_id"
+  }]
+  devices_yaml_file = "devices.yaml"
+}
+# tftest:modules=1:resources=2
+
+```
+
+## Example integrated with Data Foundation Platform
+In this example, we will show how to extend **[Data Foundations Platform](../../data-solutions/data-platform-foundations/)** including IoT Platform as a new source of data. 
+
+![Target architecture](./diagram_iot.png)
+
+1. First, we will setup Environment following instructions in **[Environment Setup](../../data-solutions/data-platform-foundations/01-environment/)** to setup projects and SAs required. Get output variable project_ids.landing as will be used later
+
+1. Second, execute instructions in **[Environment Setup](../../data-solutions/data-platform-foundations/02-resources/)** to provision PubSub, DataFlow, BQ,... Get variable landing-pubsub as will be used later to create IoT Registry
+
+1. Now it is time to provision IoT Platform. Modify landing-project-id and landing_pubsub_topic_id with output variables obtained before. Create device certificates as shown in the Simple Example and register them in devices.yaml file together with deviceids.
+
+```hcl
+module "iot-platform" {
+  source     = "./iot-core"
+  project_id = "landing-project-id"
+  region = "europe-west1"
+  telemetry_pub_sub_topic_id = "landing_pubsub_topic_id"
+  status_pub_sub_topic_id = "status_pubsub_topic_id"
+  devices_yaml_file = "devices.yaml"
+}
+# tftest:modules=1:resources=2
```

Comment:
Make sure to update these numbers once you fix the path above. Same with the other examples.